temporalio/workflow_service.py

```diff
@@ -0,0 +1,279 @@
+            wsv1.RespondActivityTaskCanceledByIdResponse,
+        )
+        self.respond_activity_task_completed = self._new_call(
+            "respond_activity_task_completed",
+            wsv1.RespondActivityTaskCompletedRequest,
+            wsv1.RespondActivityTaskCompletedResponse,
+        )
+        self.respond_activity_task_completed_by_id = self._new_call(
+            "respond_activity_task_completed_by_id",
+            wsv1.RespondActivityTaskCompletedByIdRequest,
+            wsv1.RespondActivityTaskCompletedByIdResponse,
+        )
+        self.respond_activity_task_failed = self._new_call(
+            "respond_activity_task_failed",
+            wsv1.RespondActivityTaskFailedRequest,
+            wsv1.RespondActivityTaskFailedResponse,
+        )
+        self.respond_activity_task_failed_by_id = self._new_call(
+            "respond_activity_task_failed_by_id",
+            wsv1.RespondActivityTaskFailedByIdRequest,
+            wsv1.RespondActivityTaskFailedByIdResponse,
+        )
+        self.respond_query_task_completed = self._new_call(
+            "respond_query_task_completed",
+            wsv1.RespondQueryTaskCompletedRequest,
+            wsv1.RespondQueryTaskCompletedResponse,
+        )
+        self.respond_workflow_task_completed = self._new_call(
+            "respond_workflow_task_completed",
+            wsv1.RespondWorkflowTaskCompletedRequest,
+            wsv1.RespondWorkflowTaskCompletedResponse,
+        )
+        self.respond_workflow_task_failed = self._new_call(
+            "respond_workflow_task_failed",
+            wsv1.RespondWorkflowTaskFailedRequest,
+            wsv1.RespondWorkflowTaskFailedResponse,
+        )
+        self.scan_workflow_executions = self._new_call(
+            "scan_workflow_executions",
+            wsv1.ScanWorkflowExecutionsRequest,
+            wsv1.ScanWorkflowExecutionsResponse,
+        )
+        self.signal_with_start_workflow_execution = self._new_call(
+            "signal_with_start_workflow_execution",
+            wsv1.SignalWithStartWorkflowExecutionRequest,
+            wsv1.SignalWithStartWorkflowExecutionResponse,
+        )
+        self.signal_workflow_execution = self._new_call(
+            "signal_workflow_execution",
+            wsv1.SignalWorkflowExecutionRequest,
+            wsv1.SignalWorkflowExecutionResponse,
+        )
+        self.start_workflow_execution = self._new_call(
+            "start_workflow_execution",
+            wsv1.StartWorkflowExecutionRequest,
+            wsv1.StartWorkflowExecutionResponse,
+        )
+        self.terminate_workflow_execution = self._new_call(
+            "terminate_workflow_execution",
+            wsv1.TerminateWorkflowExecutionRequest,
+            wsv1.TerminateWorkflowExecutionResponse,
+        )
+        self.update_namespace = self._new_call(
+            "update_namespace",
+            wsv1.UpdateNamespaceRequest,
+            wsv1.UpdateNamespaceResponse,
+        )
```

Comment:
Seems like this sort of repetition should be easy to eliminate in Python

Reply:
Will this approach support autocomplete when users try to use this class?

Reply:
Absolutely - that is the main reason I am limited on how much metaprogramming I can do